runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -64,7 +64,10 @@ thread_local static RTDevice *RTD_PTR = nullptr;
 /**
  * @brief get the active device.
  */
-auto getQuantumDevicePtr() -> QuantumDevice * { return RTD_PTR->getQuantumDevicePtr(); }
+auto getQuantumDevicePtr() -> const std::unique_ptr<QuantumDevice> &
```

Comment:
I don't know if we should have something that is part of C++ in the C-API. How about also obtaining a destructor pointer?

Reply:
The CAPI implementation is compiling with a C++ compiler and there shouldn't be any issues with using C++ stdlibs for non extern-c functions. are you aware of any?